src/realm/object-store/shared_realm.cpp

```diff
@@ -254,7 +254,11 @@ void Realm::read_schema_from_group_if_needed()
         }
     }
     else {
-        ObjectStore::verify_valid_external_changes(m_schema.compare(schema, m_config.schema_mode));
+        if (!m_config.is_schema_additive()) {
```

Comment:
I don't think we need to perform this check if the schema is additive (this is the reason why client reset test was failing). I would not matter if there are differences in terms of tables, since the whole schema will be returned. @tgoyne

Reply:
We do need this check. This is to make it so that if another process does something like delete a table while we have the Realm file open we fail immediately with a useful error rather than crashing in mysterious ways when we try to use the table that no longer exists.

Reply:
In additive mode, we would append the deleted table either way. Anyway, I can revert this check and avoid the check and see if I can fix the test or find another alternative.

Reply:
Would this not be a problem in case of additive schemas? In additive mode, schemas will always be merged, right?